crates/nomination/src/tests.rs

```diff
@@ -1,302 +1,25 @@
-            vault: arc_vault,
-        }
-    }
-}
-
-impl SlashingAccessors<u128, FixedI128, SlashingError> for SimpleNominator {
-    fn get_slash_per_token(&self) -> Result<FixedI128, SlashingError> {
-        self.vault.lock().unwrap().get_slash_per_token()
-    }
-
-    fn get_collateral(&self) -> u128 {
-        self.collateral
-    }
-
-    fn mut_collateral<F>(&mut self, func: F) -> Result<(), SlashingError>
-    where
-        F: Fn(&mut u128) -> Result<(), SlashingError>,
-    {
-        func(&mut self.collateral)
-    }
-
-    fn get_total_collateral(&self) -> Result<u128, SlashingError> {
-        self.vault.lock().unwrap().get_total_collateral()
-    }
-
-    fn mut_total_collateral<F>(&mut self, func: F) -> Result<(), SlashingError>
-    where
-        F: Fn(&mut u128) -> Result<(), SlashingError>,
-    {
-        self.vault.lock().unwrap().mut_total_collateral(func)
-    }
-
-    fn get_backing_collateral(&self) -> Result<u128, SlashingError> {
-        self.vault.lock().unwrap().get_backing_collateral()
-    }
-
-    fn mut_backing_collateral<F>(&mut self, func: F) -> Result<(), SlashingError>
-    where
-        F: Fn(&mut u128) -> Result<(), SlashingError>,
-    {
-        self.vault.lock().unwrap().mut_backing_collateral(func)
-    }
-
-    fn get_slash_tally(&self) -> FixedI128 {
-        self.slash_tally
-    }
-
-    fn mut_slash_tally<F>(&mut self, func: F) -> Result<(), SlashingError>
-    where
-        F: Fn(&mut FixedI128) -> Result<(), SlashingError>,
-    {
-        func(&mut self.slash_tally)
-    }
-}
+use mocktopus::mocking::*;
 
 #[test]
-fn test_non_vaults_cannot_become_operators() {
+fn should_not_deposit_against_invalid_vault() {
     run_test(|| {
         assert_err!(
-            Nomination::opt_in_to_nomination(Origin::signed(BOB)),
-            TestError::VaultNotFound
+            Nomination::_deposit_collateral(ALICE, BOB, 100),
+            TestError::VaultNotOptedInToNomination
         );
     })
 }
 
-fn nominate_slash_nominate(
-    arc_vault: Arc<Mutex<SimpleVault>>,
-    nominator1: &mut SimpleNominator,
-    nominator2: &mut SimpleNominator,
-    vault_deposit: u128,
-    nominator1_deposit: u128,
-    nominator2_deposit: u128,
-    slash: u128,
-) {
-    let backing_collateral_at_slash = vault_deposit + nominator1_deposit;
-    assert_ok!(arc_vault.lock().unwrap().try_deposit_collateral(vault_deposit));
-    assert_ok!(nominator1.try_deposit_collateral(nominator1_deposit));
-    assert_ok!(arc_vault.lock().unwrap().slash_collateral(slash));
-    assert_ok!(nominator2.try_deposit_collateral(nominator2_deposit));
-    assert_ok!(
-        arc_vault.lock().unwrap().get_backing_collateral(),
-        backing_collateral_at_slash - slash + nominator2_deposit
-    );
-    assert_ok!(
-        arc_vault.lock().unwrap().compute_collateral(),
-        vault_deposit - slash * vault_deposit / backing_collateral_at_slash - 1
-    );
-}
-
-#[test]
-fn test_nomination_slash_should_be_correct() {
```

Comment:
Why delete all the tests?

Reply:
They are no longer relevant since adopting the `staking` pallet which does this accounting.